frontend/server/src/main/java/org/pytorch/serve/metrics/LogMetric.java

```diff
@@ -0,0 +1,76 @@
+package org.pytorch.serve.metrics;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.pytorch.serve.util.ConfigManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class LogMetric extends IMetric {
+    /**
+     * Note: hostname, timestamp, and requestid(if available) are automatically added in log metric.
+     */
+    private static final Logger loggerTsMetrics =
+            LoggerFactory.getLogger(ConfigManager.MODEL_SERVER_METRICS_LOGGER);
+
+    private static final Logger loggerModelMetrics =
+            LoggerFactory.getLogger(ConfigManager.MODEL_METRICS_LOGGER);
+
+    public LogMetric(
+            MetricBuilder.MetricType type, String name, String unit, List<String> dimensionNames) {
+        super(type, name, unit, dimensionNames);
+    }
+
+    @Override
+    public void addOrUpdate(List<String> dimensionValues, double value) {
+        // Used for logging frontend metrics
+        String metricString = this.buildMetricString(dimensionValues, null, value);
+        loggerTsMetrics.info(metricString);
+    }
+
+    @Override
+    public void addOrUpdate(List<String> dimensionValues, String requestIds, double value) {
+        // Used for logging backend metrics
+        String metricString = this.buildMetricString(dimensionValues, requestIds, value);
+        loggerModelMetrics.info(metricString);
+    }
+
+    private String buildMetricString(
```

Comment:
can you have two buildMetricString so that if requestid condition can be removed?
- one have requestid => callled by modelMetrics.
- the other one doesn't have requestid => called by tsMetrics.